library/spdm_requester_lib/libspdm_req_set_key_pair_info.c

```diff
@@ -0,0 +1,213 @@
+    if (LIBSPDM_STATUS_IS_ERROR(status)) {
+        return status;
+    }
+    LIBSPDM_ASSERT (message_size >= transport_header_size);
+    spdm_response = (void *)(message);
+    spdm_response_size = message_size;
+
+    status = libspdm_receive_spdm_response(
+        spdm_context, session_id, &spdm_response_size, (void **)&spdm_response);
+    if (LIBSPDM_STATUS_IS_ERROR(status)) {
+        goto receive_done;
+    }
+
+    /* -=[Validate Response Phase]=- */
+    if (spdm_response_size < sizeof(spdm_message_header_t)) {
+        status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+        goto receive_done;
+    }
+    if (spdm_response->header.spdm_version != spdm_request->header.spdm_version) {
+        status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+        goto receive_done;
+    }
+    if (spdm_response->header.request_response_code == SPDM_ERROR) {
+        status = libspdm_handle_error_response_main(
+            spdm_context, session_id,
+            &spdm_response_size,
+            (void **)&spdm_response, SPDM_SET_KEY_PAIR_INFO, SPDM_SET_KEY_PAIR_INFO_ACK);
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            goto receive_done;
+        }
+    } else if (spdm_response->header.request_response_code != SPDM_SET_KEY_PAIR_INFO_ACK) {
+        status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+        goto receive_done;
+    }
+
+    spdm_response_size = sizeof(spdm_set_key_pair_info_ack_response_t);
+
+    status = LIBSPDM_STATUS_SUCCESS;
+
+    /* -=[Log Message Phase]=- */
+    #if LIBSPDM_ENABLE_MSG_LOG
+    libspdm_append_msg_log(spdm_context, spdm_response, spdm_response_size);
+    #endif /* LIBSPDM_ENABLE_MSG_LOG */
+
+receive_done:
+    libspdm_release_receiver_buffer (spdm_context);
+    return status;
+}
+
+libspdm_return_t libspdm_set_key_pair_info(void *spdm_context, const uint32_t *session_id,
+                                           uint8_t key_pair_id,
+                                           uint8_t operation,
+                                           uint16_t desired_key_usage,
+                                           uint32_t desired_asym_algo,
+                                           uint8_t desired_assoc_cert_slot_mask
+                                           )
+{
+    libspdm_context_t *context;
+    size_t retry;
+    uint64_t retry_delay_time;
+    libspdm_return_t status;
+
+    context = spdm_context;
+    context->crypto_request = true;
+    retry = context->retry_times;
+    retry_delay_time = context->retry_delay_time;
+    do {
+        status = libspdm_try_set_key_pair_info(context, session_id, key_pair_id,
+                                               operation,
+                                               desired_key_usage,
+                                               desired_asym_algo,
+                                               desired_assoc_cert_slot_mask);
+        if ((status != LIBSPDM_STATUS_BUSY_PEER) || (retry == 0)) {
```

Comment:
Don't need the `retry == 0` anymore. See https://github.com/DMTF/libspdm/pull/2815.

Reply:
Good catch. Fixed.